### PR TITLE
Move common code for loading/deleting configuration to config.go

### PR DESCRIPTION
**Pull request by kkaneda**
I guess we can do more code unification, but doing that in an incremental fashion.

### CODE REVIEW

File: server/config.go [hunk 1]
```diff
@@ -229,3 +234,70 @@ func RunSetPerm(ctx *Context, keyPrefix, configFileName string) {
...
+
+// GetConfig retrieves the configuration for the specified key. If the
+// key is empty, all configurations are returned. Otherwise, the
+// leading "/" path delimiter is stripped and the configuration
+// matching the remainder is retrieved. Note that this will retrieve
+// the default config if "key" is equal to "/", and will list all
+// configs if "key" is equal to "". The body result contains a listing
+// of keys and retrieval of a config. The output format is determined
+// by the request header.
+func GetConfig(db *client.KV, configPrefix proto.Key, config gogoproto.Message, path string, r *http.Request) (body []byte, contentType string, err error) {
```
Comment (petermattis) [inline server/config.go]: Go style doesn't have an explicit line length limit, but I like to stay around 100 characters to make side-by-side diffs more readable. I'd probably put a newline before the `path string` parameter.

**Comment by petermattis** [comment]
LGTM, though I agree with @jamesgraves that it looks like these functions could be private to the server package.

File: server/config.go (hunk 1 above)
Comment (kkaneda) [inline server/config.go]: Sure, I've added a newline as you suggested. Will keep the style in mind.

**Comment by kkaneda** [comment]
(For some reason, James' comment didn't appear in the email thread.. )

@jamesgraves , @petermattis , you're right. I've changed the functions to private.

**Pull request merged**